packages/e2e/test/artillery/StakePoolSearch.ts

```diff
@@ -0,0 +1,155 @@
+ * Used to track time of first request to measure cache efficiency
+ */
+let firstUser = '';
+
+/**
+ * Gets the name of the stat to measure depending on the filter this user is using to perform the query
+ *
+ * @param ctx the artillery context
+ * @returns the name of the stat we are measuring
+ */
+const getStatName = (ctx: ArtilleryContext<StakePoolSearchVars>) => {
+  const { _uid, vars } = ctx;
+
+  if (_uid === firstUser) return 'query.First';
+
+  const { identifier, status } = vars.args.filters!;
+  const nameStatId = identifier ? 'id' : '__'; // ATM id is the only identifier used by the test
+  const statusStatName = status && status.length > 0 ? 'status' : '______';
+
+  return `query.${nameStatId}_${statusStatName}`;
+};
+
+const randomizeFilter = (args: QueryStakePoolsArgs) => {
+  // Apply a random filter to 80% of the remaining users in the list, or if it's empty.
+  if (poolsId.length === 0 || Math.random() > 0.2) {
+    args.filters!.status = [
+      [
+        Cardano.StakePoolStatus.Activating,
+        Cardano.StakePoolStatus.Active,
+        Cardano.StakePoolStatus.Retired,
+        Cardano.StakePoolStatus.Retiring
+      ][Math.floor(Math.random() * 4)]
+    ];
+  }
+
+  // We can't perform calls with id until we have some ids;
+  // when we have at least one id, we want to perform 80% calls with id and 20% without
+  if (poolsId.length > 0 && Math.random() > 0.2) {
+    // If we have data to randomize: let's do it
+    args.filters!.identifier = {
+      values: [{ id: poolsId[Math.floor(Math.random() * poolsId.length)] }]
+    };
+  }
+};
+
+export const healthCheck: FunctionHook<StakePoolSearchVars> = async (ctx, ee, done) => {
+  const { vars } = ctx;
+
+  vars.healthCheckArray = [];
+
+  try {
+    const result = await provider.healthCheck();
+
+    if (result.ok) vars.healthCheckArray = [1];
+    else {
+      ee.emit('counter', 'healthCheck.unhealthy', 1);
+      logger.warn('Not ok healthCheck result', result);
+    }
+  } catch (error) {
+    ee.emit('counter', 'healthCheck.error', 1);
+    logger.error('Error thrown while performing healthCheck', error);
+  }
+
+  done();
+};
+
+export const performQuery: FunctionHook<StakePoolSearchVars> = async (ctx, ee, done) => {
+  const { vars, _uid } = ctx;
+  const { pageResults } = vars;
+  let { args } = vars;
+
+  // If not already initialized: prepare the new request starting from the first page
+  if (!args) {
+    args = { filters: {}, pagination: { limit: PAGE_SIZE, startAt: 0 } };
+    vars.args = args;
+  }
+
+  // If we have a pageResults, this is not the first call of the session: request next page
+  if (pageResults) args.pagination!.startAt += PAGE_SIZE;
+  // This is the first call of the first virtual user, we need to make a query that will fully prime the API cache
+  else if (firstUser === '') firstUser = _uid;
+  // This is the first call of the session for all users but the first one: randomize query parameters
+  else randomizeFilter(args);
```

Comment:
```suggestion
  else {
    args.filter = randomizeFilter(args);
  }
```

Reply:
Our lint rules should probably reflect this

Reply:
It's about procedural vs functional, not {}

Reply:
Sorry @mkazlauskas ,

if this is for a matter of readability, simply adding a `return args.filter` at the end of `randomizeFilter` (which results in a useless assignment) would be enough.
If the core of the request is to change `randomizeFilter` in order to make a copy of `args.filter`, modifying it rather than the original, and returning it; I think the amount of the additional resources cost from the change would not worth.

Reply:
It's about modifying the argument object. In general treating arguments as read-only makes it easier to reason about a function: arguments=inputs, return=outputs. I think it is a good practice to use this functional style instead of procedural.

> I think the amount of the additional resources cost from the change would not worth.
Why would it use significantly more resources? It can simply return `typeof args.filters` (there's probably an interface for that). You could then just do `{ filters: randomizeFilter(poolIds) }` [here](https://github.com/input-output-hk/cardano-js-sdk/pull/371/files#diff-f69af2c64d445d76a60faebdd225486f0997596e8ac518387fa57d2e237ad898R115)

Reply:
1185fc810 should solve this